core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -596,6 +596,95 @@ class DeltaSuite extends QueryTest
     }
   }
 
+  test("batch write: append, dynamic partition overwrite") {
+    withTempDir { tempDir =>
+      def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
+
+      Seq(1, 2, 3).toDF
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("append")
+        .save(tempDir.getCanonicalPath)
+
+      Seq(1, 5).toDF
+        .withColumn("part", $"value" % 2)
+        .write
+        .format("delta")
+        .partitionBy("part")
+        .mode("overwrite")
+        .option("partitionOverwriteMode", "dynamic")
+        .save(tempDir.getCanonicalPath)
+      checkDatasetUnorderly(data.toDF.select($"value".as[Int]), 1, 2, 5)
+
+      Seq(("a", "x"), ("b", "y"), ("c", "x")).toDF("value", "part")
```

Comment:
Can we make this a separate `withTempDir { ... }` and not overwrite? I think it's cleaner

Reply:
i saw you suggested adding withTempDir twice but i think only one made sense? the way it is now do you still want to add one?

Reply:
Yep what you did is what I meant. Looks good, sorry if was confusing